src/classes/InventoryManager.ts

```diff
@@ -39,18 +40,33 @@ export default class InventoryManager {
         }
 
         let genericCheck = generics;
+        let genericIndex = -1; // index of a generic match
         // if we looking at amount we can trade and the sku is a generic unusual, always set generic to true
         const isGenericSku = /^[0-9]*;5$/.test(sku);
         if (isGenericSku) {
             genericCheck = true;
         }
 
+        const gSku = SKU.fromString(sku);
+        gSku.effect = null;
+        if (generics) {
+            // figure out if we even have a generic sku
+            // Index of of item in Pricelist
+            genericIndex = this.pricelist.getIndex(null, gSku);
+        }
+        const normalIndex = this.pricelist.getIndex(sku);
+
         // Pricelist entry
-        const match = genericCheck ? this.pricelist.getPrice(sku, true, true) : this.pricelist.getPrice(sku, true);
+        const match =
+            genericCheck && genericIndex !== -1 && normalIndex === -1
+                ? this.pricelist.getPrice(SKU.fromObject(gSku), true, true)
+                : this.pricelist.getPrice(sku, true);
 
         // Amount in inventory should only use generic amount if there is a generic sku
         const amount =
-            genericCheck && match ? this.inventory.getAmountOfGenerics(sku, true) : this.inventory.getAmount(sku, true);
+            genericCheck && match && genericIndex !== -1
+                ? this.inventory.getAmountOfGenerics(SKU.fromObject(gSku), true)
+                : this.inventory.getAmount(sku, true);
```

Comment:
looks good to me, but I wonder why that `normalIndex` is not here as well.

Reply:
because let's say this is `false` (due to `normalIndex !== -1`)
https://github.com/TF2Autobot/tf2autobot/blob/fc585b3a395affcaef5d77c4d15634525565f3f0/src/classes/InventoryManager.ts#L61
It should then read this
https://github.com/TF2Autobot/tf2autobot/blob/fc585b3a395affcaef5d77c4d15634525565f3f0/src/classes/InventoryManager.ts#L63

but the `this.pricelist.getPrice(sku, true)` return non-null result (which mean non-generic exist in the pricelist).

Then, when it read this
https://github.com/TF2Autobot/tf2autobot/blob/fc585b3a395affcaef5d77c4d15634525565f3f0/src/classes/InventoryManager.ts#L67
it will be `true` and use `getAmountOfGenerics` instead of `getAmount` for non-generic.

Correct me if I am wrong.

Reply:
oh wait, it can be `false` due to `genericIndex === -1`. right?

Reply:
The first statement says, only pull a generic sku if we first, have a generic sku and second, we don't have a specific sku for the item. IE, generic sku never overrides price settings for specific.

The second statement says, only use the "generic can buy amount" if there is a generic sku present. The second really fixes the present bug because prior it was always trying to use the generic amount even if a generic sku didn't exist so it would get hits on other particle types even though there wasn't a generic sku.

Reply:
got it. Thanks for the reply. I will merge this soon.